src/dbnode/storage/index/block.go

```diff
@@ -543,15 +545,20 @@ func (b *block) queryWithSpan(
 		// Ensure that the block contains any of the relevant time segments for the query range.
 		doc := iter.Current()
```

Comment:
This whole range-check code block has gotten pretty big so maybe split to another func. Then we can also defer / make more readable the cleanup func?

Reply:
Good call. Refactored this a bit. PTAL